torch_geometric/utils/rag/graph_store.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Dict, Optional
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import FeatureStore
+from torch_geometric.distributed import LocalGraphStore
+from torch_geometric.sampler import (
+    BidirectionalNeighborSampler,
+    NodeSamplerInput,
+    SamplerOutput,
+)
+from torch_geometric.typing import EdgeTensorType
+from torch_geometric.utils import index_sort
+
+
+class NeighborSamplingRAGGraphStore(LocalGraphStore):
+    """Neighbor sampling based graph-store to store & retrieve graph data."""
+    def __init__(  # type: ignore[no-untyped-def]
+        self,
+        feature_store: Optional[FeatureStore] = None,
```

Comment:
Why is the feature_store an input to the graph store?
Can't we just initialize the sampler outside the graph store?